src/main/java/ch/kalunight/zoe/repositories/LeagueAccountRepository.java

```diff
@@ -0,0 +1,317 @@
+      result = query.executeQuery(finalQuery);
+
+      List<DTO.LeagueAccount> accounts = Collections.synchronizedList(new ArrayList<>());
+      int rowCount = result.last() ? result.getRow() : 0;
+      if(rowCount == 0) {
+        return accounts;
+      }
+      result.first();
+      while(!result.isAfterLast()) {
+        accounts.add(new DTO.LeagueAccount(result));
+        result.next();
+      }
+
+      return accounts;
+    }finally {
+      RepoRessources.closeResultSet(result);
+    }
+  }
+
+  public static List<DTO.LeagueAccount> getLeaguesAccountsWithGameCardsId(long gameCardsId) throws SQLException {
+    ResultSet result = null;
+    try (Connection conn = RepoRessources.getConnection();
+        Statement query = conn.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);) {
+
+      String finalQuery = String.format(SELECT_LEAGUES_ACCOUNTS_WITH_GAME_INFO_CARD_ID, gameCardsId);
+      result = query.executeQuery(finalQuery);
+
+      List<DTO.LeagueAccount> accounts = Collections.synchronizedList(new ArrayList<>());
+      int rowCount = result.last() ? result.getRow() : 0;
+      if(rowCount == 0) {
+        return accounts;
+      }
+      result.first();
+      while(!result.isAfterLast()) {
+        accounts.add(new DTO.LeagueAccount(result));
+        result.next();
+      }
+
+      return accounts;
+    }finally {
+      RepoRessources.closeResultSet(result);
+    }
+  }
+
+  public static void createLeagueAccount(long playerId, Summoner summoner, String server) throws SQLException {
+    try (Connection conn = RepoRessources.getConnection();
+        Statement query = conn.createStatement();) {
+
+      String finalQuery = String.format(INSERT_LEAGUE_ACCOUNT, playerId, summoner.getName(), summoner.getId(),
+          summoner.getAccountId(), summoner.getPuuid(), server);
+      query.execute(finalQuery);
+    }
+  }
+
+  public static List<DTO.LeagueAccount> getLeaguesAccounts(long guildId, long discordPlayerId) throws SQLException {
+    ResultSet result = null;
+    try (Connection conn = RepoRessources.getConnection();
+        Statement query = conn.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);) {
+
+      String finalQuery = String.format(SELECT_LEAGUES_ACCOUNTS_WITH_GUILDID_AND_PLAYER_DISCORD_ID, guildId, discordPlayerId);
+      result = query.executeQuery(finalQuery);
+
+      List<DTO.LeagueAccount> accounts = Collections.synchronizedList(new ArrayList<>());
+      int rowCount = result.last() ? result.getRow() : 0;
+      if(rowCount == 0) {
+        return accounts;
+      }
+      result.first();
+      while(!result.isAfterLast()) {
+        accounts.add(new DTO.LeagueAccount(result));
+        result.next();
+      }
+
+      return accounts;
+    }finally {
+      RepoRessources.closeResultSet(result);
+    }
+  }
+
+  public static List<DTO.LeagueAccount> getAllLeaguesAccounts(long guildId) throws SQLException {
+    ResultSet result = null;

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Always check the return of one of the navigation method (next,previous,first,last) of a ResultSet.](https://app.codacy.com/manual/KaluNight/Zoe-Discord-Bot/pullRequest?prid=4739924)